dev/tests/agent.rs

```diff
@@ -271,6 +271,82 @@ async fn spawn_with_agent() -> Result<()> {
     Ok(())
 }
 
+struct SpawnTestContext {
+    nats: Nats,
+    agent: Agent,
+    req: SpawnRequest,
+}
+
+impl SpawnTestContext {
+    async fn proxy_route(self) -> Option<String> {
+        self.agent
+            .db
+            .get_proxy_route(self.req.backend_id.id())
+            .await
+            .expect("Expected proxy route.")
+    }
+
+    async fn connection(self) -> TypedNats {
+        self.nats
+            .connection()
+            .await
+            .expect("expected nats connection")
+    }
+
+    async fn state_subscription(Arc(self)) -> BackendStateSubscription {
```

Comment:
`self` is syntax sugar for `self: Self`, but I don't think it works through an `Arc`; you need to de-sugar it to `self: Arc<Self>`